include/deal.II/base/point.h

```diff
@@ -89,6 +89,28 @@ template <int dim, typename Number = double>
 class Point : public Tensor<1,dim,Number>
 {
 public:
+
+  /**
+   * Comparison functor struct to compare two Points and return if one is
+   * "less" than the other one. This can be used to use Point<dim> as a key in
+   * std::map.
+   *
+   * Comparison is done by comparing values in each dimension in ascending
+   * order (first x, then y, etc.). Note that comparisons are done without an
+   * epsilon, so points need to have identical floating point components to be
+   * considered equal.
+   */
+  struct ComparisonHelper
```

Comment:
I don't like this class in `Point`. The reason is that it suggests that points can be ordered, but we know that this is not true. If you do need this, define it in the place where you need it.

I'm going to note that we used the same comparison function for DoFTools::map_support_points_to_dofs, which now is simply templatized on the `Comp` data type because it led to all sorts of numerical instabilities.